packages/models/src/resource-binding/manager.ts

```diff
@@ -0,0 +1,168 @@
+import { Block, Header, Output, Input, HexString, utils, OutPoint } from '@ckb-lumos/base'
+import { BI } from '@ckb-lumos/bi'
+import { Actor, ActorMessage, ActorURI, MessagePayload } from '..'
+import { TypescriptHash, LockscriptHash } from './types'
+import { ResourceBindingRegistry, ResourceBindingManagerMessage } from './interface'
+import { outpointToOutPointString } from './utils'
+import { Listener } from '@kuai/io'
+import { OutPointString } from '..'
+import type { Subscription } from 'rxjs'
+import { ChainSource } from '@kuai/io/lib/types'
+
+export class Manager extends Actor<object, MessagePayload<ResourceBindingManagerMessage>> {
+  #registry: Map<TypescriptHash, Map<LockscriptHash, ResourceBindingRegistry>> = new Map()
+  #registryOutpoint: Map<OutPointString, ResourceBindingRegistry> = new Map()
+  #registryReverse: Map<ActorURI, [TypescriptHash, LockscriptHash]> = new Map()
+  #lastBlock: Block | undefined = undefined
+  #topBlockNumber = BI.from(0)
+
+  constructor(private _listener: Listener<Header>, private _dataSource: ChainSource) {
+    super()
+  }
+
+  onListenBlock = (blockHeader: Header) => {
+    const currentBlockNumber = BI.from(blockHeader.number)
+    if (currentBlockNumber.gt(this.topBlockNumber)) {
+      this.topBlockNumber = currentBlockNumber
+    }
+  }
+
+  private update(pollingInterval = 1000) {
+    return setInterval(async () => {
+      if (this.#topBlockNumber.gt(0) && this.#topBlockNumber.gt(BI.from(this.#lastBlock?.header.number ?? 0))) {
+        this.updateStore(await this._dataSource.getBlock(this.#topBlockNumber.toHexString()))
+      }
+    }, pollingInterval)
+  }
+
+  private updateStore(block: Block) {
```

Comment:
What if update cells in a batch, as

1. get all inputs in the block
2. get all outputs in the block
3. filter outputs that consumed in the block
4. filter inputs that consume an output in the block
5. remove remaining inputs and add remaining outputs

Reply:
> What if update cells in a batch, as
> 
> 1. get all inputs in the block
> 2. get all outputs in the block
> 3. filter outputs that consumed in the block
> 4. filter inputs that consume an output in the block
> 5. remove remaining inputs and add remaining outputs

I just don't well understand. Does it mean to filter the input cells and output cells by the `Store`s and fill them in one message for each instance of `Store`?

Reply:
> 

OK, I see, this is better. I will improve it by this way.

Reply:
@yanguoyu To achieve this, I think the `Store` may be added a new message payload to support to update outputs and remove inputs together. Maybe like this:

```typescript
interface Message {
  type: 'update_cells',
  update: {
    witness: HexString,
    cell: Cell
  }[],
  remove: OutPointString[]
}
```

Is that ok?

Reply:
I think we can send two types of messages like this.
```typescript
export type StoreMessage =
  | {
      type: 'remove_cell'
      value: OutPointString[]
    }
  | {
      type: 'update_cell'
      value: UpdateStorageValue | UpdateStorageValue[]
    }

```

Reply:
I have made manager to send two messages to each store when block came, one for update outputs and one for remove inputs. 

Tests about this will be committed today.

Reply:
> > > What if update cells in a batch, as
> > > 
> > > 1. get all inputs in the block
> > > 2. get all outputs in the block
> > > 3. filter outputs that consumed in the block
> > > 4. filter inputs that consume an output in the block
> > > 5. remove remaining inputs and add remaining outputs
> > 
> > 
> > I just don't well understand. Do you mean to filter the input cells and output cells by the `Store`s and fill them in one message for each instance of `Store`?
> 
> Similarly, the logic in the PR is to handle cells one by one from the perspective of a transaction, it would write/update the state multiple times. What if we consider the cells affected from the perspective of a block, some cells are destroyed and some cells are generated in the block.
> 
> Once cells are inspected from the block's level, the most brilliant point is that many `internal changes` in a block could be ignored, e.g., sequential transactions.
> 
> in block #n
> 
> 1. Cell_A -> | tx_1 | -> Cell_B
> 2. Cell_B -> | tx_2 | -> Cell_C
> 3. Cell_C -> | tx_3 | -> Cell_D
> 
> Only Cell_A is eliminated and Cell_D is created.
> 
> An analogy in front-end is the reconciliation in react, multiple actions would take place in a frame, but only the last state will be rendered.
> 
> `Update cells` instead of `update a cell` is of course a good improvement.

I have a simple question about this.
Could tx1 and tx2 be in the same block? If so, I think that Cell_B has not been generated onchain while sending the tx2. Because the Cell_B will be generated when in tx1, which will be executed in the same block with tx2.

Reply:
The example shows transactions in one block. Although `Cell_B` is consumed immediately, it still has a record on the chain.

A similar technology called `lightning network` discards the intermediate status and only the `start status` and `end status` are submitted on the chain.

Ref: https://lightning.network/

Reply:
> The example shows transactions in one block. Although `Cell_B` is consumed immediately, it still has a record on the chain.
> 
> A similar technology called `lightning network` discards the intermediate status and only the `start status` and `end status` are submitted on the chain.

My understanding is that Cell_A and Cell_B can't be in the same tx, but they can be in the same block, i.e. the output of tx1 of a block can be the input of tx2, right?

Reply:
I'm sorry to tell you it will change to this
```typescript
{ type: 'update_cells`, value: UpdateStorageValue[] }
```